networkx/drawing/layout.py

```diff
@@ -1121,6 +1122,236 @@ def multipartite_layout(G, subset_key="subset", align="vertical", scale=1, cente
+        Controls the  "pull" to  the center  of mass  of the
+        plot (0,0)
+    adjust_sizes: bool (default: False)
+        Prevent node overlapping in the layout
+    dissuade_hubs : bool (default: false)
+        Prevent hub clustering
+    edge_weight_influence : bool (default: False)
+        Generate layout with or without considering the edge
+        weights
+    linlog : bool (default: False)
+        Use log attraction rather than linear attraction
+    dim: number (default: 2)
+       Sets the dimensions of the layout. This parameter is ignored if pos is given.
+
+    Examples
+    --------
+    >>> import networkx as nx
+    >>> G = nx.florentine_families_graph()
+    >>> nx.draw(G, pos=nx.forceatlas2_layout(G))
+
+    [1] https://journals.plos.org/plosone/article/file?id=10.1371/journal.pone.0098679&type=printable
+
+    """
+    import numpy as np
+
+    def estimate_factor(n, swing, traction, speed, speed_efficiency, jitter_tolerance):
+        """
+        ForceAtlas2 helper function
+        Computes scaling factor for force
+        """
+
+        # estimate jitter
+        opt_jitter = 0.05 * np.sqrt(n)
+        min_jitter = np.sqrt(opt_jitter)
+        max_jitter = 10
+        min_speed_efficiency = 0.05
+
+        other = min(max_jitter, opt_jitter * traction / n**2)
+        jitter = jitter_tolerance * max(min_jitter, other)
+
+        if swing / traction > 2.0:
+            if speed_efficiency > min_speed_efficiency:
+                speed_efficiency *= 0.5
+            jitter = max(jitter, jitter_tolerance)
+        if swing == 0:
+            target_speed = np.inf
+        else:
+            target_speed = jitter * speed_efficiency * traction / swing
+
+        if swing > jitter * traction:
+            if speed_efficiency > min_speed_efficiency:
+                speed_efficiency *= 0.7
+        elif speed < 1000:
+            speed_efficiency *= 1.3
+
+        max_rise = 0.5
+        speed = speed + min(target_speed - speed, max_rise * speed)
+        return speed, speed_efficiency
+
+    # parse optional pos positions
+    if pos is None:
+        pos = nx.random_layout(G, dim=dim)
+    else:
+        dim = len(next(iter(pos.values())))
+
+    # Scale the position of the node proportional to the @pos
+    max_pos_range = 1
+    min_pos_range = 0
+    # check if we have a valid pos else just return (empty graph)
+    if pos:
+        max_pos_range = np.array([max(i) for i in pos.values()]).max()
+        min_pos_range = np.array([min(i) for i in pos.values()]).min()
+    else:
+        return pos
+
+    # center locs around 0
+    pos_arr = (np.random.rand(len(G), dim) - 0.5) * (max_pos_range - min_pos_range)
+
+    mass = np.zeros(len(G))
+    size = np.zeros(len(G))
+    for idx, node in enumerate(G.nodes()):
+        if node in pos:
+            pos_arr[idx] = pos[node].copy()
+        mass[idx] = G.nodes[node].get("mass", G.degree(node) + 1)
+        size[idx] = G.nodes[node].get("size", 1)
```

Comment:
Just for clarification do you mean that we should pass mass and size as an input parameter, leave it out or document it?

Reply:
I think it's probably easiest to leave it out at this stage - it can always be added in a follow-up PR.

Reply:
In that case I vote to document it and leave it in. The size and mass are some key attributes that one could use to center on a particular cluster for example. Having separate parameters as inputs as a list, say, "feels" a bit heavy give the size of the inputs of the layout.

Reply:
IMV it really has to be made configurable. Even if it's documented, hard-coded attribute names introduce inflexibility that can catch users by surprise. For example, it wouldn't be unreasonable for a user to have a `size` attribute on the node intended to indicate the size of the glyph used in the actual visualization. To me, this feels like a feature that violates the principal of least surprise.

Reply:
I don't think we should worry about the inputs being "heavy". From a memory perspective they aren't heavy [O(nodes) not O(edges) and storing on the graph takes at least twice the memory as storing in a list or array] and from a code-effort perspective making the list/array, it should be easier for anyone who hasn't stored it on the graph and as easy as `list(nx.get_node_attributes(G, "mass").values())` if they have stored it on the graph.

Since most of our layout/drawing functions pass node and edge info in via input parameters it is probably better to do that here too. But I agree with @rossbar that unless there is a good reason to vary these values (and there apparently isn't since it isn't advertised in any way) it is probably better to just use the default values for these parameters.

Reply:
Apologies for the confusion. I wasn't referring to heavy in the sense of some computational or memory load. I was referring to that the input to the function seems "a lot" compared to other layouts.  I have pushed some changes indicating its use.